packages/webpack-extensive-lodash-replacement-plugin/index.js

```diff
@@ -332,74 +335,90 @@ const LODASH_MODULE_NAMES = [
 	'zipWith',
 ];
 
-function throwError( message, error = null ) {
-	throw new Error(
-		`[ExtensiveLodashReplacementPlugin] ${ message }${ error ? ' Error: ' : '' }${ error }`
+function createError( message, error = null ) {
+	return new Error(
+		`[ExtensiveLodashReplacementPlugin] ${ message }${ error ? ` Error: ${ error }` : '' }`
 	);
 }
 
-class ExtensiveLodashReplacementPlugin {
-	constructor( baseFile = './package.json' ) {
-		this.baseFile = path.resolve( baseFile );
-		let data;
+const resolverOptions = {
+	fileSystem: new CachedInputFileSystem( new NodeJsInputFileSystem(), 4000 ),
+	extensions: [ '.js' ],
+	resolveToContext: true,
+};
 
-		try {
-			data = fs.readFileSync( this.baseFile );
-		} catch ( error ) {
-			throwError( 'Could not read root package.json.', error );
-		}
+function getModuleForPath( rootPath, packageName ) {
+	const moduleResolver = ResolverFactory.createResolver( resolverOptions );
```

Comment:
You don't need to import and use `enhanced-resolver` explicitly. The `NormalModuleFactory` (the `nmf` param passed to you hook) has a `getResolver` method:
```js
const moduleResolver = nmf.getResolver( 'normal' );
moduleResolver.resolve( .... );
```
The resolver is already configured with all the `resolveOptions` and file systems: all dependencies are resolved using this one.

Reply:
Oh, I didn't know about that! Thanks, that's a lot simpler 🙂